packages/bigtable/src/row.js

```diff
@@ -158,25 +123,58 @@ util.inherits(Row, common.GrpcServiceObject);
  * // }
  */
 Row.formatChunks_ = function(chunks) {
-  var families = [];
-  var chunkList = [];
+  var rows = [];
+  var familyName;
+  var qualifierName;
+
+  chunks.reduce(function(row, chunk) {
+    var family;
+    var qualifier;
+
+    row.data = row.data || {};
 
-  chunks.forEach(function(chunk) {
-    if (chunk.resetRow) {
-      chunkList = [];
+    if (chunk.rowKey) {
+      row.key = Mutation.convertFromBytes(chunk.rowKey);
     }
 
-    if (chunk.rowContents) {
-      chunkList.push(chunk.rowContents);
+    if (chunk.familyName) {
+      familyName = chunk.familyName.value;
+    }
+
+    if (familyName) {
+      family = row.data[familyName] = row.data[familyName] || {};
```

Comment:
Yep!

Let's assuming that `row.data[familyName]` is null.

```js
var row = {
  data: {
    familyName: null
  }
};

var familyExtended = extend(row.data[familyName], {});

assert(row.data[familyName]); // throws
assert(familyExtended); // does not throw
```

Because `row.data[familyName]` is returning null, it's not extending a reference, but rather creating a new object and storing it in a variable.